pkg/proxy/runner/bounded_frequency_runner.go

```diff
@@ -136,18 +131,17 @@ func (bfr *BoundedFrequencyRunner) Loop(stop <-chan struct{}) {
 			bfr.minIntervalTimer.Stop()
 			bfr.maxIntervalTimer.Stop()
 
-			// do not allow to reset the retryInterval during the execution
-			bfr.retryMu.Lock()
-			defer bfr.retryMu.Unlock()
-			// stop but do not reset it, it is a one shot timer,
-			// only rearmed when calling Retry().
-			bfr.retryIntervalTimer.Stop()
-			bfr.retrying = false
-
 			// avoid crashing if the function executed crashes
 			defer utilruntime.HandleCrash()
-			bfr.fn()
-
+			err := bfr.fn()
+			// an error will schedule a retry after the retryInterval,
+			// any successful run until that period will stop the retry attempt.
+			if err != nil {
```

Comment:
Maybe have a global `var RetryError` and only retry if `errors.Is(err, RetryError)`?

Reply:
we are not going to expose this and the usage is mostly by historical reasons, do we plan to do something if the error is not retryable?